workspaces/src/network/transaction.rs

```diff
@@ -0,0 +1,159 @@
+use near_primitives::transaction::{
+    Action, AddKeyAction, CreateAccountAction, DeleteAccountAction, DeleteKeyAction,
+    DeployContractAction, FunctionCallAction, StakeAction, TransferAction,
+};
+use near_primitives::views::FinalExecutionOutcomeView;
+
+use crate::rpc::client::{
+    send_batch_tx_and_retry, Client, DEFAULT_CALL_DEPOSIT, DEFAULT_CALL_FN_GAS,
+};
+use crate::types::{AccessKey, AccountId, Balance, Gas, InMemorySigner, PublicKey};
+
+use super::CallExecutionDetails;
+
+#[derive(Debug, Clone)]
+pub struct CallArgs {
+    pub function: String,
+    pub args: Vec<u8>,
+    pub deposit: Balance,
+    pub gas: Gas,
+}
+
+impl CallArgs {
+    pub fn new(function: &str) -> Self {
+        Self {
+            function: function.into(),
+            args: Vec::new(),
+            deposit: DEFAULT_CALL_DEPOSIT,
+            gas: DEFAULT_CALL_FN_GAS,
+        }
+    }
+
+    pub fn args(mut self, args: Vec<u8>) -> Self {
+        self.args = args;
+        self
+    }
+
+    pub fn args_json<U: serde::Serialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = serde_json::to_vec(&args)?;
+        Ok(self)
+    }
+
+    pub fn args_borsh<U: borsh::BorshSerialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = args.try_to_vec()?;
+        Ok(self)
+    }
+
+    pub fn deposit(mut self, deposit: u128) -> Self {
+        self.deposit = deposit;
+        self
+    }
+
+    pub fn gas(mut self, gas: u64) -> Self {
+        self.gas = gas;
+        self
+    }
+}
+
+impl From<CallArgs> for Action {
+    fn from(args: CallArgs) -> Self {
+        Self::FunctionCall(FunctionCallAction {
+            method_name: args.function,
+            args: args.args,
+            deposit: args.deposit,
+            gas: args.gas,
+        })
+    }
+}
+
+pub struct Transaction<'a> {
+    client: &'a Client,
+    signer: InMemorySigner,
+    receiver_id: AccountId,
+    actions: Vec<Action>,
+}
```

Comment:
Shouldn't this be called something like `TransactionBuilder` to stay consistent with `CallBuilder`? Also, expanding on the topic above, `BatchCallBuilder` might be more suitable?

Reply:
yeah you're right that it should be consistent, but `CallBuilder` should be renamed to `CallTransaction` instead, since every other API such as near-api-js and near-primitives defines this `Transaction` type where you add actions. So think its best to stay consistent with those and just rename `CallBuilder` instead